### PR TITLE
feature: Handle trial debugger errors

### PR DESCRIPTION
**Pull request by Youakeem**
## Describe your changes
1. Co-locate trial debugger types and utils with the route
2. Move the trial contract form to own component
3. Manage the form state with `useFormState`
4. Return error message instead of throwing in server action
5. Show error messages under the submit button

## Justify why they are needed
We are getting a generic error screen when something happen on the server. To better communicate the errors we are now showing them on the screen after submission.

## Checklist before requesting a review

- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by Youakeem** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4076?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4076** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4076?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4075** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4075?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4074** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4074?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Youakeem and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by guilhermespopolin — APPROVED** [review]
Nice one :)

File: apps/store/src/app/debugger/trial/components/TrialContractForm.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    .toISOString()
+    .split('T')[0]
+
+  const partnerOptions = [
+    { name: 'Avy', value: 'AVY' },
+    { name: 'Stena Fastigheter', value: 'STENA_FASTIGHETER' },
+    { name: 'Byggvesta', value: 'BYGGVESTA' },
+    { name: 'Samtrygg', value: 'SAMTRYGG' },
+    { name: 'Hedvig', value: 'HEDVIG' },
+  ]
```
[the remark this answers is not in the document]
Comment (Youakeem) [reply apps/store/src/app/debugger/trial/components/TrialContractForm.tsx:27]: Yeah, moving `partnerOptions` outside the component is a good idea. 

I am not sure about `useFormState` initial state tho, I like that it's close to where it's being used and we might want to use something related to a component like a prop or something coming from a hook in it.

**Comment by Youakeem** [comment]
### Merge activity

* **Mar 19, 11:25 AM EDT**: @Youakeem started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4076).

**Pull request merged**